state-synchronizer/src/coordinator.rs

```diff
@@ -189,84 +204,124 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         }
     }
 
-    async fn request_sync(&mut self, request: SyncRequest) {
-        self.known_version = self
+    /// Sync up coordinator state with the local storage.
+    async fn sync_state_with_local_storage(&mut self) {
```

Comment:
Should return a Result. The caller would `expect` only upon startup / start sync request.